numba/np/arraymath.py

```diff
@@ -3699,12 +3699,16 @@ def searchsorted_inner(a, v, v_last, lo, hi, n):
         .. [1] https://github.com/numpy/numpy/blob/809e8d26b03f549fd0b812a17b8a166bcd966889/numpy/core/src/npysort/binsearch.cpp#L173
         """  # noqa: E501
         if np.isnan(v):
-            # Find the first nan (i.e. the last from the end of a,
-            # since there shouldn't be many of them in practice)
-            for i in range(n, 0, -1):
-                if not np.isnan(a[i - 1]):
-                    return i
-            return 0
+            if side_is_right:
```

Comment:
Could you add a comment explaining the reason for `side_is_right`?

Reply:
@guilhermeleobas I have updated this PR now that the `searchsorted` changes have been merged - if you get a mo, it should be in the reviewable state.